packages/object/src/index.ts

```diff
@@ -235,9 +273,17 @@ export class DRPObject implements IDRPObject {
 		return newState;
 	}
 
-	private _setState(vertex: Vertex) {
-		const newState = this._computeState(vertex.dependencies, vertex.operation);
-		this.states.set(vertex.hash, { state: newState });
+	// biome-ignore lint: values can be anything
+	private _setState(vertex: Vertex, state?: Map<string, any>) {
+		if (state) {
+			this.states.set(vertex.hash, { state });
+		} else {
+			const newState = this._computeState(
+				vertex.dependencies,
+				vertex.operation,
+			);
+			this.states.set(vertex.hash, { state: newState });
+		}
```

Comment:
```suggestion
		if (state) {
			this.states.set(vertex.hash, { state });
			return;
		}
		const newState = this._computeState(
			vertex.dependencies,
			vertex.operation,
		);
		this.states.set(vertex.hash, { state: newState });
```

Reduced Nesting, Readability